### PR TITLE
New invoicing system

### PR DESCRIPTION
**Pull request by blazejbujala**

### CODE REVIEW

File: src/main/java/com/fc/invoicing/db/Database.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.fc.invoicing.db;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface Database<T> {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/db/Database.java]: This is not needed in new Project

File: src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.fc.invoicing.db;
+
+import com.fc.invoicing.model.Company;
+import java.util.UUID;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface JpaCompanyRepository extends PagingAndSortingRepository<Company, UUID> {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java]: Just CompanyRepository now

File: src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.fc.invoicing.db;
+
+import com.fc.invoicing.model.Invoice;
+import java.util.UUID;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface JpaInvoiceRepository extends PagingAndSortingRepository<Invoice, UUID> {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java]: Just InvoiceRepository now

File: src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.fc.invoicing.db;
+
+import com.fc.invoicing.model.Company;
+import java.util.UUID;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java]: Annotation not needed

File: src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.fc.invoicing.db;
+
+import com.fc.invoicing.model.Invoice;
+import java.util.UUID;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java]: Annotation not needed

File: src/main/java/com/fc/invoicing/services/CompanyService.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            updatedCompany.setCompanyId(id);
+            return companyMapper.toDto(jpaCompanyRepository.save(companyMapper.toEntity(updatedCompany)));
+        } else {
+            throw new ItemNotFoundExemption("No company with id: " + id + " please add new company");
+        }
+    }
+
+    @Override
+    public boolean delete(UUID id) {
+        if (jpaCompanyRepository.findById(id).isPresent()) {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/services/CompanyService.java]: There is no need for this check - deleteById will throw an exception if item is not in db. https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#deleteById-ID-

File: src/main/java/com/fc/invoicing/services/CompanyService.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Override
+    public CompanyDto add(CompanyDto company) {
+        company.setCompanyId(UUID.randomUUID());
+        return companyMapper.toDto(jpaCompanyRepository.save(companyMapper.toEntity(company)));
+    }
+
+    @Override
+    public CompanyDto getById(UUID id) {
+        if (jpaCompanyRepository.findById(id).isPresent()) {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/services/CompanyService.java]: try to simplify it with jpaCompanyRepository.findById(id).map().orElseThrow()

File: src/main/java/com/fc/invoicing/services/InvoiceService.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class InvoiceService implements Database<Invoice> {
```
Comment (radoslawdabrowski) [inline src/main/java/com/fc/invoicing/services/InvoiceService.java]: implements Database is not needed here anymore

Comment (radoslawdabrowski) [reply src/main/java/com/fc/invoicing/services/InvoiceService.java]: This is just service, should not have nothing in common with db

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/fc/invoicing/services/CompanyService.java (hunk 6 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/services/CompanyService.java]: Thnak you for comment, but I've wanted  to make the method to throw exception with comment: ItemNotFoundExemption("No invoice with id: " + id), when there is on id.

File: src/main/java/com/fc/invoicing/db/Database.java (hunk 1 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/db/Database.java]: deleted

File: src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java (hunk 2 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java]: changed

File: src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java (hunk 3 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java]: changed

File: src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java (hunk 4 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/db/JpaCompanyRepository.java]: deleted

File: src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java (hunk 5 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/db/JpaInvoiceRepository.java]: deleted

File: src/main/java/com/fc/invoicing/services/InvoiceService.java (hunk 8 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/services/InvoiceService.java]: changed

File: src/main/java/com/fc/invoicing/services/CompanyService.java (hunk 7 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/services/CompanyService.java]: changed

File: src/main/java/com/fc/invoicing/services/CompanyService.java (hunk 6 above)
Comment (blazejbujala) [reply src/main/java/com/fc/invoicing/services/CompanyService.java]: changed

**Review by radoslawdabrowski — APPROVED** [review]

**Review by stanleyszysz — APPROVED** [review]

**Pull request merged**